src/TraceEvent/ETWTraceEventSource.cs

```diff
@@ -449,7 +469,17 @@ private void Initialize(string fileOrSessionName, TraceEventSourceType type)
             // Open all the traces
             for (int i = 0; i < handles.Length; i++)
             {
-                handles[i] = TraceEventNativeMethods.OpenTrace(ref logFiles[i]);
+                if (string.IsNullOrEmpty(this.etwModulePath))
+                {
+                    handles[i] = TraceEventNativeMethods.OpenTrace(ref logFiles[i]);
+                }
+                else
+                {
+                    var procAddress = this.GetFunctionPointerForEtwLibrary("OpenTraceW");
+                    var openTrace = (TraceEventNativeMethods.OpenTraceDelegate)Marshal.GetDelegateForFunctionPointer(procAddress, typeof(TraceEventNativeMethods.OpenTraceDelegate));
```

Comment:
You should cache this delegate in a field so you don't have to LoadLibrary+GetProcAddress each time around the loop.